src/collection.jl

```diff
@@ -0,0 +1,100 @@
+# helper function to match extension
+# this will be removed when there is a compat entry of endswith with regex compatibility
+function match_extension(str::String, ext::Regex)
+    r = deepcopy(ext)
+    if !endswith(r.pattern, "\$")
+        r = Regex(r.pattern * "\$", r.compile_options, r.match_options)
+    end
```

Comment:
do we need to deepcopy if you're creating a new copy of Regex? Or vice-versa? I feel like we could get away with
```julia
r = deepcopy(ext)
if !endswith(r.pattern, "\$")
    r.pattern. *= "\$"
end
```

Also, wouldn't it be worth it to wrap the whole thing in an `if` block for `VERSION < v"1.2"`, at least for now?

Reply:
Ya, an if-else statement would be good! The code you suggested above also works fine!

Reply:
Some tests fail for this version of `match_extension`

Reply:
Simply changing `pattern` doesn't change the compiled regexp, that's why I used the constructor.

Reply:
Can you explain a bit about what exactly is `compile_options` and `match_options`?

Reply:
No because I don't know _exactly_ what they are :slightly_smiling_face:  I assume `match_options` encodes the options like case-sensitivity etc.

Reply:
ah makes, sense! In that case, we can still elude the deepcopy, I believe.

Reply:
Since `Regex` is a mutable struct I wanted to avoid mutating the input object.